tests/helpers/test_translation.py

```diff
@@ -206,3 +205,39 @@ async def test_get_translation_categories(hass):
             hass, "en", "device_automation", None, True
         )
         assert "component.light.device_automation.action_type.turn_on" in translations
+
+
+async def test_translation_merging(hass, caplog):
+    """Test we merge translations of two integrations."""
+    hass.config.components.add("sensor.moon")
+    hass.config.components.add("sensor.season")
+    hass.config.components.add("sensor")
+
+    translations = await translation.async_get_translations(hass, "en", "state")
+
+    assert "component.sensor.state.moon__phase.first_quarter" in translations
+    assert "component.sensor.state.season__season.summer" in translations
+
+    # Merge in some bad translation data
+    integration = Mock(file_path=pathlib.Path(__file__))
+    hass.config.components.add("sensor.bad_translations")
+
+    with patch.object(
+        translation, "component_translation_path", return_value=mock_coro("bla.json")
```

Comment:
`component_translation_path` is a callback that returns a string. I don't understand why we patch it with a coroutine.

Reply:
Legacy, and then copy/paste. Removed them all now.